codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ProtocolUtils.java

```diff
@@ -160,4 +162,15 @@ public static void writeSafeMemberAccessor(
             consumer.accept(operand);
         });
     }
+
+    /**
+     * Determines whether a given shape will use a pointer when the shape is used as a union value.
+     *
+     * @param shape the shape to check
+     * @return true if the shape should use pointers
+     */
+    public static boolean usesPointerWhenUnionValue(Shape shape) {
+        return !(shape instanceof SimpleShape) || shape.isBlobShape() || shape.hasTrait(EnumTrait.class)
+                || shape.hasTrait(StreamingTrait.class);
+    }
```

Comment:
I'm potentially reading this wrong, but would this conditional make `struct`, `blob` and `enum`, and `streaming` shape value member types pointers? Or is this an artifact how the code writer generates list/maps?

Reply:
whoops you're right i think i have this backwards